aten/src/ATen/native/mobile/cpu/internal/Allocator.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <ATen/native/mobile/cpu/internal/Common.h>
+
+#ifdef USE_XNNPACK
+
+namespace at {
+namespace native {
+namespace mobile {
+namespace cpu {
+namespace internal {
+
+class Allocator final : public at::Allocator {
```

Comment:
This class deserves a pretty good comment explaining why you need another allocator for xnnpack.

Reply:
Will do.

On a different note, currently I am allocating output tensors with this allocator even though XNNPACK out-of-bound accesses the input only.  The rationale is that these output operators will be subsequently used as input on the next operator.

Having said that, this is obviously not complete.  This misses the case of any input tensor re-reouted to XNNPACK not allocated with this allocator - including the very first input, or any input coming from another codepath that does not use this allocator.  This means that we need the logic to copy the input tensor on demand.

I think ideally we need a way to be able to detect what allocator a given tensor was allocated with.  In that scenario, I can at least avoid re-allocating the input if it was already allocated with this allocator.

What do you think?  Is that possible currently?

Reply:
You can detect the used allocator using `cast_context`. See https://github.com/pytorch/pytorch/pull/30870 for a recent example of this.

Reply:
XNNPACK can read up to XNN_EXTRA_BYTES from output tensors too in some cases (e.g. Max Pooling operator with large kernel)

Reply:
Aah good to know, thanks for letting me know.